English/Grades/Middle.md

```diff
@@ -1,43 +1,47 @@
 # Middle C++
 
-## Кто это?
+## Who is it?
 
-Это разработчик, который понимает технический контекст разработки и способен создать дизайн и решения для функционала внутри компонента/приложения даже в случае неполноты требований. Также имеет практический опыт работы на проектах и в рамках принятых бизнес-процессов.
+It's a developer that understands the technical context of development and has abilities to create a design and solution for functionality that is a part of an application or component. The design also can be created even in case of an insufficient amount of requirements. This person also has a commercial experience background and is familiar with common business processes of development.
 
-В основном решает технические задачи, но, в отличие от джуна, способен делать это самостоятельно или под менторством синьора/тимлида.
+In general, the middle developer solves technical tasks. In comparison with a Junior, this person can do work without any help or minor assistance from a Senior/Lead engineer.
 
-## Что ожидается по умению написания кода?
 
-- Компилятор и язык его больше не пугают и практически не приносят сюрпризов, а если и приносят, то способен самостоятельно генерировать гипотезы, проверять их и копать вглубь
-- Ориентируется в базовых концепциях языка и понимает, какие ещё языки существуют и чем они отличаются
-- Пишет понятный и поддерживаемый код
-- Знает базовые принципы дизайна и на их основе способен принять техническое решение
-- Понимает не только язык программирования, но и его технический контекст, то есть понимает весь технический цикл, через который проходит код и ориентируется в инструментах, которые помогают этот цикл поддерживать:
-    - Написание кода (IDE, текстовые редакторы, практики программирования, качество кода)
-    - Хранение исходного кода и продуктов (системы контроля версий, пакетные менеджеры, серверы)
-    - Компиляция (компиляторы, системы сборки, библиотеки)
-    - Тестирование (фреймворки, стратегии тестирования)
-    - Доставка
-    - Выполнение на целевой системе
-- Глубже понимает и знает базовую информатику (структуры данных, конечные автоматы, алгоритмы)
+## What coding abilities are expected?
 
-## Что ожидается по общим навыкам?
+- The compiler and programming language is not a "magic box" anymore. Any obstacles or surprises can be solved by generating of hypothesis, validation, and confirmation/rejection.
+- Understands foundation concepts of C++, knows about other languages, and can compare them with each other
+- Writes readable, extendable, and maintainable code 
+- Knows design patterns and principles, can make technical decisions
+- Understands a technical context of the language: e.g. how the code is compiled, knows tools that help to maintain a code lifecycle:
+    - Code writing (IDE, text editors, code quality, best practices, etc.)
+    - Archivation of source code and a product (version system control, package managers, servers, etc.)
```

Comment:
Archivation(?) of source code and ~~a~~ product -> Source code and product storage